src/metal.cs

```diff
@@ -3614,109 +4094,417 @@ partial interface MTLRenderCommandEncoder : MTLCommandEncoder {
 		[Abstract]
+#else
+		[NoMac, NoiOS, TV (16,0), NoMacCatalyst, NoWatch]
+#endif
+
+#else
+		[Mac (12, 0), iOS (15, 0), TV (16, 0), MacCatalyst (15, 0), NoWatch]
 #endif
-		[Mac (12, 0), iOS (15, 0), NoTV, MacCatalyst (15, 0), NoWatch]
 		[Export ("setTileAccelerationStructure:atBufferIndex:")]
 		void SetTileAccelerationStructure ([NullAllowed] IMTLAccelerationStructure accelerationStructure, nuint bufferIndex);
 
 #if NET
+#if !TVOS
+		[Mac (12, 0), iOS (15, 0), NoTV, MacCatalyst (15, 0), NoWatch]
 		[Abstract]
+#else
+		[NoMac, NoiOS, TV (16,0), NoMacCatalyst, NoWatch]
+#endif
+
+#else
+		[Mac (12, 0), iOS (15, 0), TV (16, 0), MacCatalyst (15, 0), NoWatch]
 #endif
-		[Mac (12, 0), iOS (15, 0), NoTV, MacCatalyst (15, 0), NoWatch]
 		[Export ("setTileIntersectionFunctionTable:atBufferIndex:")]
 		void SetTileIntersectionFunctionTable ([NullAllowed] IMTLIntersectionFunctionTable intersectionFunctionTable, nuint bufferIndex);
 
 #if NET
+#if !TVOS
+		[Mac (12, 0), iOS (15, 0), NoTV, MacCatalyst (15, 0), NoWatch]
 		[Abstract]
+#else
+		[NoMac, NoiOS, TV (16,0), NoMacCatalyst, NoWatch]
+#endif
+
+#else
+		[Mac (12, 0), iOS (15, 0), TV (16, 0), MacCatalyst (15, 0), NoWatch]
 #endif
-		[Mac (12, 0), iOS (15, 0), NoTV, MacCatalyst (15, 0), NoWatch]
 		[Export ("setTileIntersectionFunctionTables:withBufferRange:")]
 		void SetTileIntersectionFunctionTables (IMTLIntersectionFunctionTable [] intersectionFunctionTable, NSRange range);
 
 #if NET
+#if !TVOS
+		[Mac (12, 0), iOS (15, 0), NoTV, MacCatalyst (15, 0), NoWatch]
 		[Abstract]
+#else
+		[NoMac, NoiOS, TV (16,0), NoMacCatalyst, NoWatch]
+#endif
+
+#else
+		[Mac (12, 0), iOS (15, 0), TV (16, 0), MacCatalyst (15, 0), NoWatch]
 #endif
-		[Mac (12, 0), iOS (15, 0), NoTV, MacCatalyst (15, 0), NoWatch]
 		[Export ("setTileVisibleFunctionTable:atBufferIndex:")]
 		void SetTileVisibleFunctionTable ([NullAllowed] IMTLVisibleFunctionTable functionTable, nuint bufferIndex);
 
 #if NET
+#if !TVOS
+		[Mac (12, 0), iOS (15, 0), NoTV, MacCatalyst (15, 0), NoWatch]
 		[Abstract]
+#else
+		[NoMac, NoiOS, TV (16,0), NoMacCatalyst, NoWatch]
+#endif
+
+#else
+		[Mac (12, 0), iOS (15, 0), TV (16, 0), MacCatalyst (15, 0), NoWatch]
 #endif
-		[Mac (12, 0), iOS (15, 0), NoTV, MacCatalyst (15, 0), NoWatch]
 		[Export ("setTileVisibleFunctionTables:withBufferRange:")]
 		void SetTileVisibleFunctionTables (IMTLVisibleFunctionTable [] functionTables, NSRange range);
+
+		[Mac (14, 0), iOS (17, 0), TV (17, 0), MacCatalyst (17, 0), NoWatch]
+#if NET
+		[Abstract]
+#endif
+		[Export ("setVertexBuffer:offset:attributeStride:atIndex:")]
+		void SetVertexBuffer ([NullAllowed] IMTLBuffer buffer, nuint offset, nuint stride, nuint index);
+
+		[Mac (14, 0), iOS (17, 0), TV (17, 0), MacCatalyst (17, 0), NoWatch]
+#if NET
+		[Abstract]
+#endif
+		[Export ("setVertexBuffers:offsets:attributeStrides:withRange:")]
+		void SetVertexBuffers (IMTLBuffer [] buffers, IntPtr offsets, IntPtr strides, NSRange range);
```

Comment:
Looks like `buffers` is a C-array: https://developer.apple.com/documentation/metal/mtlrendercommandencoder/1515987-setvertexbuffers

```suggestion
		void SetVertexBuffers (IntPtr buffers, IntPtr offsets, IntPtr strides, NSRange range);
```